app/helpers/admin/products_helper.rb

```diff
@@ -37,5 +37,9 @@ def products_return_to_url(url_filters)
 
       "#{admin_products_path}#{url_filters.empty? ? '' : "#?#{url_filters.to_query}"}"
     end
+
+    def hide_producer_column?(producer_options)
+      spree_current_user.column_preferences.products.empty? && producer_options.one?
+    end
```

Comment:
I am trying to understand this, so we hide the producer column if :
- the user has no column preference set for Bulk Edit Product page and
-  there is only one producer available ?
 
So what happen if the user set some column preference and there is only one producer available ? It looks like the user would be able to see the producer column. Which, after re checking the requirement, is correct :) 
I think we should add a comment here to clarify that, it's not super obvious from just reading the code.

Reply:
Yes, sorry my bad. It's indeed imbiguous. I've added the explaination here: https://github.com/openfoodfoundation/openfoodnetwork/pull/12854/commits/76fdf3725a88d83f278b26a8683bcf4e21375cd7